grails-app/controllers/org/pih/warehouse/reporting/ReportController.groovy

```diff
@@ -544,7 +545,14 @@ class ReportController {
     }
 
     def showInventoryByLocationReport = { MultiLocationInventoryReportCommand command ->
-        command.entries = productAvailabilityService.getQuantityOnHandByProduct(command.locations)
+        String[] categories = params.list('category').findAll { it } ?: null
+        // When accessing the page for the first time, the flag should be set to true
+        Boolean includeCategoryChildren = true
+        // If we run the report manually and the checkbox is not checked
+        if (params.button == "run" && !params.includeCategoryChildren) {
+            includeCategoryChildren = false
+        }
+        command.entries = productAvailabilityService.getQuantityOnHandByProduct(command.locations, categories, includeCategoryChildren)
```

Comment:
While I don't mind the logic being buried in getQuantityOnHandByProduct(), I think the categories + includeCategoryChildren logic could be handled with a call like this just to make it clear and reusable.
```
def categories = productService.getCategories(command.categories, includeCategoryChildren)
```
Also those two properties (categories, includeCategoryChildren) should be added to the command class.